### DISCUSSION TITLE
Integrating skywalking to java app

### DISCUSSION THREAD

**Discussion by hieu276** [apache/skywalking#9639] [Q&A] [answered]
Hi,
Skywalking APM 9.1.0
Java agent 8.12.0
Im trying to monitor my java application with skywalking and start with a simple api running on localhost (request: localhost:8001/api/hello -> response: "hello"). I embedded java agent by adding javaagent:/path/to/skywalking-package/agent/skywalking-agent.jar to JVM argument and only set service_name in agent.config file.  Only dashboard in JVM worked (1st picture) and everything else didn't worked like overview tab (2nd picture). What am i missing?

I'm so new, have a poor vocabulary and grammar, forgive me, ty.
![image](https://user-images.githubusercontent.com/48582387/191156278-91a9d3f5-456a-4e8e-b53c-db7ba7778ebb.png)
![image](https://user-images.githubusercontent.com/48582387/191157228-d930b8fe-4a3d-47ff-b51d-ef355ac3105d.png)

**Comment by wu-sheng** [comment, marked as answer]
Check supported frameworks and libraries, I think you use a way the Java agent doesn't support.

**Reply by hieu276** [reply]
i followed this setup java agent https://skywalking.apache.org/docs/skywalking-java/v8.12.0/en/setup/service-agent/java-agent/readme/
and run skywalking by starting bin/startup.bat.

**Reply by wu-sheng** [reply]
Read all docs, not just a single one. https://skywalking.apache.org/docs/skywalking-java/v8.12.0/en/setup/service-agent/java-agent/supported-list/

If you just want to try, showcase is all set for you, https://skywalking.apache.org/docs/skywalking-showcase/next/readme/

**Reply by hieu276** [reply]
You were right, i'll retry. I want to integrate skywalking to a java application deployed in non-container environment, is that necessary to setup skywalking server in container environment or just starting bin/startup.bat is enough.

**Reply by wu-sheng** [reply]
The container is not required. bat/shell is as same as container running.
